src/features/CodeLinkFeature.ts

```diff
@@ -138,6 +176,49 @@ export class LinkCodeWithSelectedNodeService {
 		);
 	}
 
+	@action.bound
+	private async linkSymbolWithSelectedNode() {
+		const lastActiveDrawioEditor =
+			this.editorManager.lastActiveDrawioEditor;
+		if (!lastActiveDrawioEditor) {
+			window.showErrorMessage("No active drawio instance.");
+			return;
+		}
+		const editor = window.activeTextEditor;
+		if (!editor) {
+			window.showErrorMessage("No text editor active.");
+			return;
+		}
+		const uri = editor?.document.uri;
+		const result = (await commands.executeCommand(
+			"vscode.executeDocumentSymbolProvider", uri
+		)) as DocumentSymbol[];
+		let items: QuickPickItem[] = [];
+		function recurse(symb: DocumentSymbol[], path: string) {
+			for (let x of symb) {
+				let curpath = path == "" ? x.name : `${path}.${x.name}`;
+				items.push(<QuickPickItem>{
+					label: `$(${symbolList[x.kind]}) ${x.name}`,
+					description: x.detail,
+					detail: curpath
+				});
+				recurse(x.children, curpath);
```

Comment:
Maybe it makes sense to consider the selection? I.e. only symbols that intersect with the current selection.

Reply:
How about a context menu entry in the editor? We can show the symbols at the cursor or selection when the command is called from the context menu (and if the command is called from the command palette, we can show every symbol)

Reply:
Hmm, it would be great if we could show that context menu entry only if a node is selected in a draw.io diagram.

Reply:
Sure, I'll add them as regular commands first then we'll see about the context menu.